### PR TITLE
[DOCS] Document ingest pipeline and processor node stats

### LINKED ISSUE #40063: [Docs] Documentation missing for ingest node metrics
IN the documentation there is a general phrase about ingest metrics under node stats. Could we document the ingest metrics returned by `GET _nodes/stats/ingest?filter_path=nodes.*.ingest`

### PR DESCRIPTION
**Pull request by jrodewig**
### Changes
- Explicitly documents ingest pipeline and processor stats returned by the node stats API.
- Adds some examples for filtering the response to only include ingest-related stats.
- Adds section headings for each metric group. I'd eventually like to put these in collapsible sections, but that's not possible for now.

Closes #40063.

It also relates to #47524, which is a broader effort to document the node stats response body and make it consistent.

### CODE REVIEW

File: docs/reference/cluster/nodes-stats.asciidoc [hunk 1]
```diff
@@ -342,24 +353,60 @@ the current running process:
...
-    The total number ingest preprocessing operations failed during the lifetime 
-    of this node.
+    (integer)
+    Total number of failed ingest operations during the lifetime of this node.
+
+`ingest.pipelines.<pipeline-id>.count`::
+    (integer)
+    Number of documents preprocessed by the ingest pipeline.
 
-On top of these overall ingest statistics, these statistics are also provided on
```
Comment (jrodewig) [inline docs/reference/cluster/nodes-stats.asciidoc]: I didn't feel this sentence was needed anymore, but let me know if the `<pipeline-id>` convention is intuitive.
I can re-add if needed.

File: docs/reference/cluster/nodes-stats.asciidoc [hunk 2]
```diff
@@ -342,24 +353,60 @@ the current running process:
...
 `ingest.total.current`::
-    The total number of documents currently being ingested.
+    (integer)
+    Total number of documents currently being ingested.
 
 `ingest.total.failed`::
-    The total number ingest preprocessing operations failed during the lifetime 
-    of this node.
+    (integer)
+    Total number of failed ingest operations during the lifetime of this node.
```
Comment (jrodewig) [inline docs/reference/cluster/nodes-stats.asciidoc]: I mostly added the datatype, removed "The" as the first word, and fixed a grammar error.

File: docs/reference/cluster/nodes-stats.asciidoc [hunk 3]
```diff
@@ -433,3 +480,31 @@ GET /_nodes/stats?groups=_all
...
+GET /_nodes/stats?metric=ingest&filter_path=nodes.*.ingest
+--------------------------------------------------
+
+To further refine the response, change the `filter_path` value.
+For example, the following request only returns ingest pipeline statistics.
+
+[source,console]
+--------------------------------------------------
+GET /_nodes/stats?metric=ingest&filter_path=nodes.*.ingest.pipelines
+--------------------------------------------------
```
Comment (jrodewig) [inline docs/reference/cluster/nodes-stats.asciidoc]: Let me know if there are other good examples to add here.

**Review by debadair — APPROVED** [review]
LGTM

**Pull request merged**